app/controllers/contactPreference/EmailVerificationController.scala

```diff
@@ -0,0 +1,134 @@
+      case _ => handleNoEmail
+    }
+  }
+
+  def checkVerificationStatus: Action[AnyContent] = (contactPreferencePredicate andThen paperPrefPredicate).async {
+    implicit user =>
+
+      implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(user.headers, Some(user.session))
+
+      extractSessionEmail match {
+        case Some(email) => emailVerificationService.isEmailVerified(email).flatMap {
+          case Some(true) => Redirect(routes.EmailVerificationController.updateContactPrefEmail())
+          case _ => emailVerificationService.createEmailVerificationRequest(
+            email,
+            routes.EmailVerificationController.updateContactPrefEmail().url
+          ).map {
+            case Some(true) =>
+              Redirect(routes.EmailVerificationController.show())
+            case Some(false) =>
+              logDebug("[EmailVerificationController][checkVerificationStatus] Email has already been verified. " +
+                "Redirecting to the update route.")
+              Redirect(routes.EmailVerificationController.updateContactPrefEmail())
+            case _ => errorHandler.showInternalServerError
+          }
+        }
+        case _ => handleNoEmail
+      }
+
+  }
+
+  def updateContactPrefEmail(): Action[AnyContent] = (contactPreferencePredicate andThen paperPrefPredicate).async {
+    implicit user =>
+
+      implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(user.headers, Some(user.session))
+      extractSessionEmail match {
+        case Some(email) => emailVerificationService.isEmailVerified(email).flatMap {
+          case Some(true) => sendUpdateRequest(email)
+          case _ =>
+            logDebug("[EmailVerificationController][checkVerificationStatus] Email has not yet been verified.")
+            Redirect(routes.EmailVerificationController.checkVerificationStatus())
+        }
+        case _ => handleNoEmail
+      }
+  }
+
+  private[controllers] def sendUpdateRequest(email: String)(implicit user: User[_]): Future[Result] = {
+    vatSubscriptionService.updateContactPrefEmail(user.vrn, email).map {
+      case Right(_) =>
+        auditService.extendedAudit(
+          ChangedEmailAddressAuditModel(
+            None,
+            email,
+            user.vrn,
+            user.isAgent,
+            None
+          ), routes.EmailVerificationController.updateContactPrefEmail().url
+        )
+        Redirect(controllers.email.routes.EmailChangeSuccessController.show())
+      case Left(ErrorModel(CONFLICT, _)) =>
+        logDebug("[ConfirmWebsiteController][updateWebsite] - There is a contact details update request " +
+          "already in progress. Redirecting user to manage-vat overview page.")
+        Redirect(appConfig.btaAccountDetailsUrl)
+      case Left(error) =>
+        logWarn(s"[ConfirmWebsiteController][updateWebsite] - ${error.status}: ${error.message}")
```

Comment:
Sorry I've just noticed this
```suggestion
        logDebug("[EmailVerificationController][sendUpdateRequest] - There is a contact details update request " +
          "already in progress. Redirecting user to BTA account details page.")
        Redirect(appConfig.btaAccountDetailsUrl)
      case Left(error) =>
        logWarn(s"[EmailVerificationController][sendUpdateRequest] - ${error.status}: ${error.message}")
```